README.md

```diff
@@ -1,3 +1,197 @@
+
+It"s composed this way:
+
+```json
+{
+  "vendorListVersion": 84,
+  "lastUpdated": "Some isodate",
+  "purposes": [
+    {
+      "id": 1,
+      "name": "Purpose name",
+      "description": "Purpose description"
+    }
+  ],
+  "features": [
+    {
+      "id": "1",
+      "name": "Feature name",
+      "description": "Feature description"
+    }
+  ],
+  "vendors": [
+    {
+      "id": "278",
+      "name": "Vendor name",
+      "policyUrl": "Vendor policy url",
+      "purposeIds": ["some", "purpose", "related", "ids"],
+      "legIntPurposeIds": ["dont", "know", "the", "aim", "of", "this"],
+      "featureIds": ["some", "feature", "ids"]
+    }
+  ]
+}
+```
+
+Here the relationships are the following:
+
+- Vendors can relate to multiple purposes
+- Vendors can own multiple features
+- A feature can concern multiple vendors
+- A purpose can contain multiple vendors
+- There is _no direct_ relationship between features and purposes.
+
+## The ConsentString
+
+The `Consent String` is a Base64 string generated from information based on the previously described `VendorList` and the different consents provided by the user.
+IAB has provided an sdk available on npm & github: https://github.com/InteractiveAdvertisingBureau/Consent-String-SDK-JS
+
+We can use it this way:
+
+```javascript
+import { ConsentString } from "consent-string";
+
+const consents = new ConsentString();
+
+/**
+ * we have to register this and be sure that our vendor list is at a correct format
+ */
+consents.setGlobalVendorList(ourGlobalVendorList);
+
+/**
+ * Giving current user consent to a specific vendor or purpose
+ */
+consents.setPurposeAllowed(1, true);
+consents.setVendorAllowed(287, true);
+```
+
+The `ConsentString` aims to be shared with the vendors that try to get user information. When these vendors will "ask" (see CMP part) for consents, this `ConsentString` will be shared with them. If they are concerned, they can collect information. Else, they can't.
+
+### Vendor based / Purpose based / Feature based
+
+It's possible to manage granularity for consent management. Some website prefers to let the user chose which `vendor` are authorized to gather information.
+Some other, like us at M6Web, prefers to use a less granular one and to manage consents at a purpose level.
+
+**Example of a vendor based consent management:**
+
+- [x] Google Analytics
+- [ ] Krux
+
+**Example of a vendor based consent management:**
+
+- [ ] Information storage and access
+- [x] Personalisation
+
+_Feature based is not available for now with IAB Consent Framework. But since the feature field is available in the VendorList, we can imagine managing consents using feature based approach_
+
+## Consent Management Provider (CMP)
+
+"CMP" means a company that can read the vendors chosen by a website operator and the consent status of an end user (either service specific (through a first-party cookie) or global (through a third-party cookie).
```

Comment:
I'm not sure we can copy / paste the text as it is from their doc. I think they are under a specific licence. We need to rephrase it a bit IMO 😊